module/os/linux/zfs/zfs_ctldir.c

```diff
@@ -590,7 +590,8 @@ struct inode *
 zfsctl_root(znode_t *zp)
 {
 	ASSERT(zfs_has_ctldir(zp));
-	igrab(ZTOZSB(zp)->z_ctldir);
+	/* Must have an existing ref, so igrab() cannot return NULL */
+	VERIFY3P(igrab(ZTOZSB(zp)->z_ctldir), !=, NULL);
```

Comment:
```suggestion
	zhold(ZTOZSB(zp)->z_ctldir);
```

Reply:
Not sure about that transform from igrab to zhold, as the zhold version would end up as igrab(ZTOI(ZTOZSB(zp)->z_ctldir)) rather than igrab(ZTOZSB(zp)->z_ctldir)